src/databricks/labs/ucx/source_code/jobs.py

```diff
@@ -139,11 +140,38 @@ def _register_spark_python_task(self, graph: DependencyGraph):
         path = WorkspacePath(self._ws, notebook_path)
         return graph.register_notebook(path)
 
-    def _register_python_wheel_task(self, graph: DependencyGraph):  # pylint: disable=unused-argument
+    @staticmethod
+    def _find_first_matching_distribution(
+        path_lookup: PathLookup, distribution_name: str
+    ) -> metadata.Distribution | None:
+        # Prepared exists in importlib.metadata.__init__pyi, but is not defined in importlib.metadata.__init__.py
+        normalize_name = metadata.Prepared.normalize  # type: ignore
+        for library_root in path_lookup.library_roots:
+            for path in library_root.glob("*.dist-info"):
+                distribution = metadata.Distribution.at(path)
+                if normalize_name(distribution.name) == normalize_name(distribution_name):
```

Comment:
```suggestion
        need_distribution = normalize_name(distribution_name)
        for library_root in path_lookup.library_roots:
            for path in library_root.glob("*.dist-info"):
                distribution = metadata.Distribution.at(path)
                if normalize_name(distribution.name) == need_distribution:
```
nit: this reduces time by O(N/2)